pkg/client/clientcmd/api/types.go

```diff
@@ -26,7 +25,11 @@ import (
 
 // Config holds the information needed to build connect to remote kubernetes clusters as a given user
 type Config struct {
-	api.TypeMeta `json:",inline"`
+	// Legacy field from pkg/api/types.go TypeMeta.
+	// TODO(jlowdermilk): remove this after eliminating downstream dependencies.
+	Kind string `json:"kind,omitempty"`
+	// Version of the schema for this config object.
+	APIVersion string `json:"apiVersion,omitempty"`
```

Comment:
From a design point of view, I think APIVersion is more important to keep than kind. Clients that consume kubeconfig generally don't consume other objects. But the practical reason for keeping both is to avoid breaking downstream clients/consumers of kubeconfig, in particular gcloud.